src/lib/kwlib.sh

```diff
@@ -714,6 +714,25 @@ get_env_name_encoded_with_pwd() {
   printf '%s-%s' "$env_name" "$encoded_pwd"
 }
 
+# Extracts the environment name from a string of the format "env_name-base64(PWD)".
+# Assumes env_name may contain hyphens.
+#
+# @encoded: The encoded string (e.g., "my-env-L2hvbWUvYXBvbG8y")
+#
+# Return:
+# Returns 0 if successful, 1 if input is empty or malformed.
+# Prints the env_name (everything before the last hyphen) if successful.
+decode_env_name_with_pwd() {
```

Comment:
from my comments in the first commit, I'm not convinced that we need this function.